### DISCUSSION TITLE
Using an image as threshold

### DISCUSSION THREAD

**Discussion by dcsobral** [ImageMagick/ImageMagick#4147] [Help] [answered]
I'm creating a mask based on the pixel value of a monochromatic image (`dtm.png -threshold $THRESHOLD`) and I want to change it to use a second image as the threshold value. So, basically, if pixel value of image 1 is less than pixel value of image 2 at the same coordinate then black otherwise white. How do I do that?

I'll eventually change it so that black is `#738cce` and white is transparent (or vice versa -- I'd just need to change image order and less than to less or equal than), so if I can avoid an extra step to do so it would be appreciated as I'm dealing with large images.

**Comment by fmw42** [comment]
One way is to use -fx, but it will be a bit slow. see https://imagemagick.org/script/fx.php

`convert image1 image2 -fx "u<v?0:1" result`

**Reply by dcsobral** [reply]
I was avoiding `-fx` because the doc said it is slow. I tested this anyway and it was an order of magnitude slower.

**Comment by snibgo** [comment, marked as answer]
Subtract image2 from image1. The result is positive where image1 > image2, or negative where image1 < image2, or zero where they are the same. Threshold at zero so negative and zero becomes zero and positive becomes maximum.

```
magick image1 image2 -compose MinusSrc -composite -threshold 0 x.png
```
I am unsure what your second paragraph means. Perhaps (Windows syntax):
```
magick image1 image2 -compose MinusSrc -composite -threshold 0 -fill #738cce -transparent White x.png
```

**Reply by dcsobral** [reply]
This is what I wanted indeed, though it is missing an `-opaque black` for the fill.

**Reply by snibgo** [reply]
Oops, sorry, yes.
